src/lib/input/input-container.ts

```diff
@@ -566,19 +578,24 @@ export class MdInputContainer implements AfterViewInit, AfterContentInit, AfterC
   private _syncAriaDescribedby() {
     if (this._mdInputChild) {
       let ids: string[] = [];
-      let startHint = this._hintChildren ?
-          this._hintChildren.find(hint => hint.align === 'start') : null;
-      let endHint = this._hintChildren ?
-          this._hintChildren.find(hint => hint.align === 'end') : null;
-
-      if (startHint) {
-        ids.push(startHint.id);
-      } else if (this._hintLabel) {
-        ids.push(this._hintLabelId);
-      }
 
-      if (endHint) {
-        ids.push(endHint.id);
+      if (this._getDisplayedMessages() === 'hint') {
+        let startHint = this._hintChildren ?
+            this._hintChildren.find(hint => hint.align === 'start') : null;
+        let endHint = this._hintChildren ?
+            this._hintChildren.find(hint => hint.align === 'end') : null;
```

Comment:
The reason this is more complicated is because there's also a `hintLabel` property that can be used to set the start hint. The extra logic is to make sure we put the start hint first (whether its from a `md-hint` or `hintLabel`, followed by the end hint)